packages/flutter/lib/src/widgets/widget_state.dart

```diff
@@ -1144,10 +1144,19 @@ class WidgetStatesController extends ValueNotifier<Set<WidgetState>> {
   /// Creates a WidgetStatesController.
   WidgetStatesController([Set<WidgetState>? value]) : super(<WidgetState>{...?value});
 
+  @override
+    Set<WidgetState> get value => <WidgetState>{...super.value};
```

Comment:
@justinmc does this remind you at all of how the TextEditingController works? I was looking for some kind of similar example for this behavior, that looks to be the only one that comes close.

Reply:
Yes, good call.

@heisenunes This probably shouldn't create a new Set on `get`, but I think your setter is probably right. TextEditingController creates a new TextEditingValue when you just set something like the `text`: 

https://github.com/flutter/flutter/blob/51ff96b9ac5bea0df1a831126dc8948e91ed46c4/packages/flutter/lib/src/widgets/editable_text.dart#L274-L280

Would your problem still be fixed if you removed this getter?